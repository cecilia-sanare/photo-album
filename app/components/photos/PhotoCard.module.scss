.card {
    position: relative;
    display: flex;
    flex-direction: column;
    gap: 10px;
    padding: 10px;
    box-sizing: border-box;
    width: 150px;
    border-radius: 3px;
    background: rgba(0, 0, 0, 0.25);
    color: white;
    cursor: pointer;

    transition: 0.15s ease-in-out;
    transition-property: background-color;

    &:hover {
        background: rgba(0, 0, 0, 0.5);
    }

    > .thumbnail {
        border-radius: inherit;
    }

    > .id {
        display: flex;
        align-items: center;
        padding: 0 10px;
        min-height: 30px;
        line-height: 1;
        background: rgba(0, 0, 0, 0.5);
        border-radius: 3px;
        position: absolute;
        top: 15px;
        left: 15px;
    }
}