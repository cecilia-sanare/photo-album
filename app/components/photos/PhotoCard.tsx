import { Photo } from '../../store/slices/photos.slice';
import * as styles from './PhotoCard.module.scss';
import { setSelectedPhoto } from '../../store/slices/search.slice';
import { useAppDispatch } from '../../store/store';

type PhotoCardProps = {
    photo: Photo;
}

export function PhotoCard({
    photo
}: PhotoCardProps) {
    const dispatch = useAppDispatch();

    return (
        <div 
            className={styles.card}
            onClick={() => dispatch(setSelectedPhoto(photo))}
        >
            <div className={styles.id}>{photo.id}</div>
            <img className={styles.thumbnail} src={photo.thumbnailUrl} />
            {photo.title}
        </div>
    )
}