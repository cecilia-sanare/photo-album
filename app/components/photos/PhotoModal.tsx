import { useCallback } from 'react';
import { Photo } from '../../store/slices/photos.slice'
import { setSelectedPhoto } from '../../store/slices/search.slice';
import { useAppDispatch } from '../../store/store';
import { Modal } from '../common/Modal'
import { ModalHeader } from '../common/ModalHeader';
import { ModalContent } from '../common/ModalContent';

type PhotoModalProps = {
  photo?: Photo;
}

export function PhotoModal({
  photo
}: PhotoModalProps) {
  const dispatch = useAppDispatch();

  const onClose = useCallback(() => {
    dispatch(setSelectedPhoto(null));
  }, [dispatch]);

  return (
    <Modal
      opened={Boolean(photo)}
      onClose={onClose}
    >
      <ModalHeader onClose={onClose}>
        {photo?.title}
      </ModalHeader>
      <ModalContent>
        <img src={photo?.url} />
      </ModalContent>
    </Modal>
  )
}
