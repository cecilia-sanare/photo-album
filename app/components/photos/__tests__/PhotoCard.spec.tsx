import { fireEvent, render } from '@testing-library/react';
import { PhotoCard } from '../PhotoCard';
import Chance from 'chance';
import { Photo } from '../../../store/slices/photos.slice';

const chance = new Chance();

describe('<Select />', () => {
  it('should render the Select', () => {
    const photo: Photo = {
      albumId: chance.integer(),
      id: chance.integer(),
      thumbnailUrl: chance.url(),
      title: chance.word(),
      url: chance.url()
    };

    const component = render(<PhotoCard photo={photo} />);

    expect(component.baseElement).toMatchSnapshot();
  });

  describe('prop(onClick)', () => {
    it('should support being clicked', async () => {
      const onClick = jest.fn();
      const photo: Photo = {
        albumId: chance.integer(),
        id: chance.integer(),
        thumbnailUrl: chance.url(),
        title: chance.word(),
        url: chance.url()
      };

      const component = render(<PhotoCard photo={photo} onClick={onClick} />);

      expect(onClick).toHaveBeenCalledTimes(0);

      await fireEvent.click(component.getByTestId('photo-card'));

      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
