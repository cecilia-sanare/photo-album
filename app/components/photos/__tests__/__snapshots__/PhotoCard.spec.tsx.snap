// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render the Select 1`] = `
<body>
  <div>
    <div
      class="card"
      data-testid="photo-card"
    >
      <div
        class="id"
      >
        1
      </div>
      <img
        class="thumbnail"
        src="http://ololacin.vn/hab"
      />
      kigasvo
    </div>
  </div>
</body>
`;
